public/app/features/scenes/scenes/demoDataNode.tsx

```diff
@@ -0,0 +1,233 @@
+
+import { Scene } from '../components/Scene';
+import { SceneFlexChild, SceneFlexLayout } from '../components/SceneFlexLayout';
+import { SceneToolbar } from '../components/SceneToolbar';
+import { VizPanel } from '../components/VizPanel';
+import { SceneDataProviderNode } from '../core/SceneDataProviderNode';
+import { SceneTimeRange } from '../core/SceneTimeRange';
+import { SceneEditManager } from '../editor/SceneEditManager';
+
+export function getFlexLayoutTest1(): Scene {
+  const timeRangeNode1 = new SceneTimeRange({
+    range: getDefaultTimeRange(),
+  });
+
+  const timeRangeNode2 = new SceneTimeRange({
+    range: getDefaultTimeRange(),
+  });
+
+  const scene = new Scene({
+    $editor: new SceneEditManager({}),
+    title: 'Flex layout test',
+    children: [
+      new SceneFlexLayout({
+        direction: 'column',
+        children: [
+          new SceneFlexLayout({
+            direction: 'column',
+            children: [
+              new SceneFlexChild({
+                size: {
+                  ySizing: 'content',
+                },
+                children: [
+                  new SceneToolbar({
+                    orientation: 'horizontal',
+                    children: [timeRangeNode1],
+                  }),
+                ],
+              }),
+              new SceneFlexChild({
+                children: [
+                  new SceneFlexLayout({
+                    direction: 'row',
+                    children: [
+                      new SceneFlexChild({
+                        children: [
+                          new SceneDataProviderNode({
+                            inputParams: {
+                              timeRange: timeRangeNode1,
+                            },
+                            queries: [
+                              {
+                                refId: 'A',
+                                datasource: {
+                                  uid: 'gdev-testdata',
+                                  type: 'testdata',
+                                },
+                                scenarioId: 'random_walk',
+                              },
+                            ],
+                            children: [
+                              new VizPanel({
+                                pluginId: 'timeseries',
+                                title: 'Title',
+                                options: {
+                                  legend: { displayMode: 'hidden' },
+                                },
+                              }),
+                            ],
+                          }),
+                        ],
+                      }),
+
+                      new SceneFlexChild({
+                        children: [
+                          new SceneDataProviderNode({
+                            inputParams: {
+                              timeRange: timeRangeNode1,
+                            },
```

Comment:
With time range no longer being a state property that you inherit from parent, feels like we lose quite a bit of simplicity and flexibility.  All scene objects will want to be able to get the current time range in its scope (And subscribe to it), manually assigning time range to all objects feels more complex.  

There are some edge case scenarios where manually specifying/subscribing to a time range that is not your own could be useful but wonder if we can handle that exception in some way without giving up on time range as inherited scoped state.